src/module/Makefile

```diff
@@ -5,23 +5,24 @@ MOD := krfx
 KRF_SYSCALL_SRCS_FAKE := $(notdir $(wildcard $M/syscalls/*.c))
 KRF_SYSCALL_OBJS_FAKE := $(KRF_SYSCALL_SRCS_FAKE:.c=.o)
 KRF_SYSCALL_OBJS = $(foreach obj,$(KRF_SYSCALL_OBJS_FAKE),syscalls/$(obj))
-
+KRF_SYSCALL_YMLS := $(wildcard codegen/syscalls/*.yml)
 ccflags-y := -DKRF_CODEGEN=1
 
 obj-m += $(MOD).o
 $(MOD)-objs := krf.o config.o syscalls.o $(KRF_SYSCALL_OBJS)
 
-all: codegen
-	make -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) modules
+all: ruby_codegen
+	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) modules
 
-.PHONY: codegen
-codegen:
+# .PHONY: codegen # Cannot be phony because phony are always run
+ruby_codegen: $(KRF_SYSCALL_YMLS) codegen/codegen
```

Comment:
This is good, but I think we can do a full dependency relation here. Maybe something like this?

```make
all: $(KRF_SYSCALL_OBJS)
    # ...

$(KRF_SYSCALL_OBJS): $(KRF_SYSCALL_YMLS)
    # ...

$(KRF_SYSCALL_YMLS): codegen


.PHONY: codegen
codegen:
   ruby codegen/codegen
```

That rough structure should give us the build improvements *and* avoid the need to drop a stamp file (which will cause rebuild issues when the specs are changed).